_posts/pl/2024-05-17-czy-wiesz-ze-mozna-uniknac-nadmiernego-uzycia-warunkow-if-przez-wykorzystanie-ternary-operator-i-lookup-table.md

```diff
@@ -0,0 +1,133 @@
+
+## Scenariusz switch-case
+
+Jedną z alternatyw zastępowania `if/else` jest użycie `switch-case`. Lepiej pasuje do koncepcji tego, co próbujemy osiągnąć.
+```javascript
+const setCarDescriptionBasedOnBrandSwitchCase = () => {
+  const brandInput = document.getElementById('switchCaseBrandInput').value;
+ 
+    let carDesc = '';
+   
+  switch (brandInput) {
+    case 'toyota':
+      carDesc = 'Toyota - Reliable and Fuel Efficient';
+      break;
+    case 'ford':
+      carDesc = 'Ford - American Toughness';
+      break;
+    case 'honda':
+      carDesc = 'Honda - Innovative and Efficient';
+      break;
+    case 'bmw':
+      carDesc = 'BMW - Ultimate Driving Machine';
+      break;
+    default:
+      carDesc = 'Unknown Car Brand';
+  }
+ 
+  document.getElementById('carDescription').innerText = carDesc;
+};
+```
+Jednak kod również nie jest zbyt czytelny i prowadzi do złożoności. Czy da się tego uniknąć? Tak! Rozważmy kolejny scenariusz.
+
+## Scenariusz lookup table
+Stwórzmy obiekt przechowujący nazwy marek samochodowych jako klucze i opisy marek jako wartości. Dzięki temu będziemy mogli uzyskać opis marki, korzystając z notacji `obiekt[klucz]`. Dodatkowo dodajemy wartość domyślną, aby obsłużyć scenariusz, kiedy klucz nie zostanie znaleziony.
+```javascript
+const carBrandsDesctipion = {
+    'toyota'   : 'Toyota - Reliable and Fuel Efficient',
+    'ford'  : 'Ford - American Toughness',
+    'honda' : 'Honda - Innovative and Efficient',
+    'bmw'   : 'BMW - Ultimate Driving Machine',
+    'default': 'Unknown Car Brand'
+};
+ 
+const setCarDescription = () => {
+  const brandName = document.getElementById('brandInput').value;
+    document.getElementById('carDescription').innerText = carBrandsDesctipion[brandName]
+        ? carBrandsDesctipion[brandName]
+        : carBrandsDesctipion['default'];
+};
+```
+Teraz mamy kod wykorzystujący lookup table i operator ternarny. Jest on znacznie zwięzły i czytelniejszy niż `if/else` i `switch-case`. Dodatkowo zwiększyliśmy łatwość utrzymania kodu, a jednocześnie zmniejszyliśmy jego złożoność, mając tylko jedno sprawdzenie logiczne dla domyślnego rozwiązania awaryjnego.
+
+Takim samym sposobem możemy obsłużyć bardziej skomplikowany przypadek wykorzystujący operatory porównania.
+
+Przykład dla `if/else`:
+```javascript
+const classifyEngine = (horsepower) => {
+  if (horsepower >= 500) {
+    return "High Performance";
+  } else if (horsepower >= 400) {
+    return "Strong";
+  } else if (horsepower >= 300) {
+    return "Moderate";
+  } else if (horsepower >= 200) {
+    return "Basic";
+  } else {
+    return "Low";
+  }
+};
+```
+Jak można zauważyć, operatory logiczne ciągle się powtarzają.
+
+Wyodrębnijmy więc nasze dane do tablicy obiektów. Dzięki temu, przeszukamy naszą tablicę w poszukiwaniu obiektu, który spełnia jeden warunek z wykorzystaniem operatora "większe lub równe".
```

Comment:
Umożliwi nam to wyszukanie w tablicy obiektu, który spełnia jeden warunek z wykorzystaniem operatora "większe lub równe".